lib/ansible/playbook/play_context.py

```diff
@@ -54,31 +55,47 @@
 # in variable names.
 
 MAGIC_VARIABLE_MAPPING = dict(
+    accelerate_port=('ansible_accelerate_port', ),
+
+    # base
     connection=('ansible_connection', ),
+    module_compression=('ansible_module_compression', ),
+    shell=('ansible_shell_type', ),
+    executable=('ansible_shell_executable', ),
+    remote_tmp_dir=('ansible_remote_tmp', ),
+
+    # connection common
     remote_addr=('ansible_ssh_host', 'ansible_host'),
     remote_user=('ansible_ssh_user', 'ansible_user'),
-    remote_tmp_dir=('ansible_remote_tmp', ),
+    password=('ansible_ssh_pass', 'ansible_password'),
     port=('ansible_ssh_port', 'ansible_port'),
+    pipelining=('ansible_ssh_pipelining', 'ansible_pipelining'),
     timeout=('ansible_ssh_timeout', 'ansible_timeout'),
-    ssh_executable=('ansible_ssh_executable', ),
-    accelerate_port=('ansible_accelerate_port', ),
-    password=('ansible_ssh_pass', 'ansible_password'),
     private_key_file=('ansible_ssh_private_key_file', 'ansible_private_key_file'),
-    pipelining=('ansible_ssh_pipelining', 'ansible_pipelining'),
-    shell=('ansible_shell_type', ),
+
+    # networking modules
     network_os=('ansible_network_os', ),
+
+    # ssh TODO: remove
```

Comment:
What's this TODO for?  It's not obvious to me (from the large commit) why these should be removed.  I use ansible_ssh_extra_args extensively.  Will this be going away?

Reply:
no, it just lives in the ssh plugin, play_context doesn't need to have it anymore